src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheckTest.java

```diff
@@ -356,6 +356,46 @@ public void testCacheWhenUrlExternalResourceContentDoesNotChange() throws Except
         verify(checker, pathToEmptyFile, pathToEmptyFile, expected);
     }
 
+    @Test
+    public void testPathRegexMatches() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("file", getResourcePath("import-control_one.xml"));
+        checkConfig.addAttribute("path", "^.*[\\\\/]src[\\\\/]test[\\\\/].*$");
+        final String[] expected = {"5:1: " + getCheckMessage(MSG_DISALLOWED, "java.io.File")};
+
+        verify(checkConfig, getPath("InputImportControl.java"), expected);
+    }
+
+    @Test
+    public void testPathRegexMatchesPartially() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("file", getResourcePath("import-control_one.xml"));
+        checkConfig.addAttribute("path", "test");
+        final String[] expected = {"5:1: " + getCheckMessage(MSG_DISALLOWED, "java.io.File")};
+
+        verify(checkConfig, getPath("InputImportControl.java"), expected);
+    }
+
+    @Test
+    public void testPathRegexDoesntMatch() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("file", getResourcePath("import-control_one.xml"));
+        checkConfig.addAttribute("path", "^.*[\\\\/]src[\\\\/]main[\\\\/].*$");
+        final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
+
+        verify(checkConfig, getPath("InputImportControl.java"), expected);
+    }
+
+    @Test
+    public void testPathRegexDoesntMatchPartially() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("file", getResourcePath("import-control_one.xml"));
+        checkConfig.addAttribute("path", "main");
```

Comment:
File name is better. I think we should still include `[\\/]`. As long as you don't write `.*` it was still be a `find` expression, right? So you don't have to completely leave out the folder structure.

This is split from my initial comment in https://github.com/checkstyle/checkstyle/pull/3724#pullrequestreview-16694125.